File: JavaScript/프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    var answer = people.length;
    
    people.sort((a, b) => a-b)
    var start = 0
    var end = people.length - 1
    
    while (start < end) {
        if (people[end] + people[start] <= limit) {
            answer -= 1
            start += 1
        }
        end -= 1
    }
    return answer;
}

File: Python3/프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    n = len(people)
    answer = n
    people.sort()
    s, e = 0, n - 1
    while s < e:
        if people[s] + people[e] <= limit:
            answer -= 1
            s += 1
        e -= 1
    return answer
